client/hotrod/src/main/java/org/infinispan/hotrod/impl/transport/netty/ByteBufUtil.java

```diff
@@ -23,9 +23,44 @@ public static byte[] readArray(ByteBuf buf) {
       return bytes;
    }
 
-   public static String readString(ByteBuf buf) {
-      byte[] strContent = readArray(buf);
-      return new String(strContent, HotRodConstants.HOTROD_STRING_CHARSET);
+   public static byte[] readMaybeArray(ByteBuf bf) {
+      int length = readMaybeVInt(bf);
+      if (length == Integer.MIN_VALUE) {
+         return null;
```

Comment:
This should check the reader index moved instead. Same issue we had in the RESP parser you mentioned.